utils/formatters/number.ts

```diff
@@ -106,11 +115,12 @@ export const formatNumber = (value: WeiSource, options?: FormatNumberOptions) =>
 	}
 
 	// specified truncation params overrides universal truncate
-	if (shouldTruncate && !truncation) {
-		if (weiValue.gt(1e6)) {
-			truncation = { divisor: 1e6, unit: 'M', decimals: 2 };
-		} else if (weiValue.gt(1e3)) {
-			truncation = { divisor: 1e3, unit: 'K', decimals: 0 };
+	if (truncateThreshold && !truncation) {
+		for (const threshold of thresholds) {
```

Comment:
I think we can use a cleaner way of doing this but can take a look in a follow up PR